modules/glean/runtime/glean-plugin.client.ts

```diff
@@ -1,21 +1,21 @@
 import Glean from '@mozilla/glean/web'
 import * as log from 'tauri-plugin-log-api'
 
-import { linkClick, pageUrl, pageView, referrerUrl } from '../../../telemetry/generated/web'
-import { userAgent } from '../../../telemetry/generated/identifiers'
-import { engagement } from '../../../telemetry/generated/ui'
-import { engagementDetails } from '../../../telemetry/engagementDetails'
+import { userAgent } from '~~/telemetry/generated/identifiers'
+import { engagement } from '~~/telemetry/generated/ui'
+import { linkClick, pageUrl, pageView, referrerUrl } from '~~/telemetry/generated/web'
+import { engagementDetails } from '~~/telemetry/engagementDetails'
 
 export default defineNuxtPlugin((nuxtApp) => {
   nuxtApp.hook('app:mounted', () => {
     log.info('Glean: App mounted, start initing glean')
 
     const GLEAN_APP_ID = 'moso-mastodon-web'
-    const env = useAppConfig().env
-    const devMode = env === ('dev' || 'canary' || 'preview')
+    const env = useBuildInfo().env
+    const devMode = env === 'dev'
     const userSettings = useUserSettings()
-    const allowGlean = getPreferences(userSettings.value, 'allowGlean')
-    const uploadEnabled = devMode && allowGlean
+    const userAllowGlean = getPreferences(userSettings.value, 'allowGlean')
+    const uploadEnabled = userAllowGlean
```

Comment:
I have removed the environment check from `uploadEnabled` because I'm not confident that stage and production will get represented correctly from either `useBuildInfo` or `useAppConfig`. In order to test this and help us understand it better, I'm passing the `channel: env` when initializing Glean - this will allow us to find out what that env gets set as in the other environments.